src/settings/s_decisions.php

```diff
@@ -147,7 +147,80 @@ function render(SettingValues $sv) {
         "</div>\n\n";
     Ht::stash_script("foldup(\$\$('cbfinal_open'),null,{f:\"c\"})");
+
+	//Share with Open Academic
+    echo "<h3 class=\"settings g\">Share with search engines</h3>\n";
+    echo "<p><b>Click the button to share accepted paper meta data with search engines</b></p>";
+    echo Ht::submit("share", "Share with search engines");
+
+    //wakes up API
+    $api = 'https://openacademicapi.azurewebsites.net/api/papers';
+    $opts = array('http'=> array('method' => 'GET'));
+    $context = stream_context_create($opts);
+    $post_result = file_get_contents($api, false, $context);
+   
+    if (isset($_REQUEST["share"])) {
+        $this->share($sv, $api);
+    }
+
 }
+    //sends https post request to the api
+    function share(SettingValues $sv, $api) {
+
+        $share = array();
+        $share["provider"] = "HotCRP";
+        $share["setId"] = "";
+        $share["venue"] = $sv->conf-> long_name;
+        $share["shortVenue"] = $sv->conf-> short_name;
+        $share["confSite"] = $sv->conf-> opt["conferenceSite"];
+        $share["version"] = "1.0";
+    
+        $papers = array();
+        $result = $sv->conf-> q_raw("SELECT paperID, title, authorInformation, abstract FROM Paper WHERE outcome > 0");
+        while (($row = edb_row($result))) {
+            $curr_p = [];
+            $curr_p['paperId'] = $row[0];
+            $curr_p['title'] = $row[1];
+            $curr_p['abstract'] = $row[3];
+            
+            //parse the authorInformation into an array
+            $authors = [];
+            $mul_authorRow = explode("\n", $row[2]);
+            
+            //the last element in the array is null, do not want the null
+            $mul_authorRow = array_slice($mul_authorRow, 0, count($mul_authorRow) - 1);
+            
+            foreach ($mul_authorRow as $sing_authorRow) {
+                $authorInfo = explode("\t", $sing_authorRow);
+                $curr_author = [];
+                //combine first name and last name to full name
+                $curr_author['name'] = $authorInfo[0]." ". $authorInfo[1];
+                if (strlen($authorInfo[2]) == 0) {
+                   $curr_author['srcId'] = $authorInfo[2];
+                } else {
+                   $curr_author['srcId'] = sha1($authorInfo[2]);
+                }
+                $curr_author['affiliation'] = $authorInfo[3];
+                $curr_author['order'] = 0;
+                array_push($authors, $curr_author);
+            }
+            $curr_p['authors'] = $authors;
+            array_push($papers, $curr_p);
+        }
+        $share["papers"] = $papers;
+
+        //sends HTTP POST REQUEST
+        $opts = array('http'=> array('method'=> 'POST', 'header'=> 'Content-type: application/json', 'content'=> json_encode($share)));
+        $context = stream_context_create($opts);
+        $post_result = file_get_contents($api, false, $context);
```

Comment:
I think you probably want to build in some authentication, so, for example, no one else can randomly submit “papers” claiming to be from hotcrp.com.

Authentication tokens are a good mechanism. There are hundreds of ways to do that.

I would prefer `https` to `http`.